server/src/main/java/org/opensearch/index/fielddata/fieldcomparator/UnsignedLongValuesComparatorSource.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.fielddata.fieldcomparator;
+
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.index.NumericDocValues;
+import org.apache.lucene.index.SortedNumericDocValues;
+import org.apache.lucene.search.DocIdSetIterator;
+import org.apache.lucene.search.FieldComparator;
+import org.apache.lucene.search.LeafFieldComparator;
+import org.apache.lucene.search.SortField;
+import org.apache.lucene.util.BitSet;
+import org.opensearch.common.Nullable;
+import org.opensearch.common.Numbers;
+import org.opensearch.common.util.BigArrays;
+import org.opensearch.index.fielddata.LeafNumericFieldData;
+import org.opensearch.index.fielddata.FieldData;
+import org.opensearch.index.fielddata.IndexFieldData;
+import org.opensearch.index.fielddata.IndexNumericFieldData;
+import org.opensearch.index.search.comparators.UnsignedLongComparator;
+import org.opensearch.search.DocValueFormat;
+import org.opensearch.search.MultiValueMode;
+import org.opensearch.search.sort.BucketedSort;
+import org.opensearch.search.sort.SortOrder;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.util.function.Function;
+
+/**
+ * Comparator source for unsigned long values.
+ *
+ * @opensearch.internal
+ */
+public class UnsignedLongValuesComparatorSource extends IndexFieldData.XFieldComparatorSource {
+
+    private final IndexNumericFieldData indexFieldData;
+    private final Function<SortedNumericDocValues, SortedNumericDocValues> converter;
+
+    public UnsignedLongValuesComparatorSource(
+        IndexNumericFieldData indexFieldData,
+        @Nullable Object missingValue,
+        MultiValueMode sortMode,
+        Nested nested
+    ) {
+        this(indexFieldData, missingValue, sortMode, nested, null);
+    }
+
+    public UnsignedLongValuesComparatorSource(
+        IndexNumericFieldData indexFieldData,
+        @Nullable Object missingValue,
+        MultiValueMode sortMode,
+        Nested nested,
+        Function<SortedNumericDocValues, SortedNumericDocValues> converter
```

Comment:
we dont use this ctor anywhere right ? I think that convertor one is needed only for longCoparatorSource since it has to convert Date/Date_NanoSeconds types to long.

Reply:
Correct, removing it